kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/domain.kt

```diff
@@ -0,0 +1,164 @@
+package io.kotest.property.arbitrary
+
+import io.kotest.property.Arb
+
+/**
+ * Generates random domains
+ *
+ * https://en.wikipedia.org/wiki/Domain_name#Domain_name_syntax
+ * https://whogohost.com/host/knowledgebase/308/Valid-Domain-Name-Characters.html
+ */
+fun Arb.Companion.domain(
+   tlds: List<String> = top120TLDs,
+   labelArb: Arb<String> = _labelArb
+): Arb<String> = arbitrary { rs ->
+   val amountOfLabels = rs.random.nextInt(1, 4)
+
+   val labels = labelArb.take(amountOfLabels).joinToString(separator = ".", postfix = ".")
+
+   labels + tlds.random(rs.random)
+}
+
+private val _labelArb = arbitrary {
+   buildString(63) {
+      append(validCharsArb.filter { it != '-' }.single(it))
+
+      append(validCharsArb.take(it.random.nextInt(0, 61), it).joinToString(""))
+
+      append(validCharsArb.filter { it != '-' }.single(it))
+
+      if(length >= 4) {
+         if(get(2) == '-' && get(3) == '-') {
+            set(3, validCharsArb.filter { it != '-' }.single(it))
+         }
+      }
+   }
+}
+
+private val validCharsArb = arbitrary { (('a'..'z') + ('A'..'Z') + ('0'..'9') + '-').random(it.random) }
+
+/**
+ * https://www.scoutdns.com/100-most-popular-tlds-by-google-index/
+ */
+private val top120TLDs = listOf(
```

Comment:
top120 ?

Reply:
From position 120 and onwards the amount of indexed pages decreases faster than from 1 to 120, that's why I cut the line at 120. But I think it could be any amount here, the spreadsheet has 1510 indexed TLDs